### ISSUE TITLE
code coverage for triggers - lines are off

### ISSUE THREAD

**Issue by jgebal** [utPLSQL/utPLSQL#231]
The DBMS_PROFILER is identifying lines of TRIGGER as they are defined in `DBA_TRIGGERS.TRIGGER_BODY`. This is usually not aligned with what is in DBA_SOURCE.
That causes two issues.
1) False-negatives are not filtered out properly on TRIGGERS
2) The covered/uncovered/not-relevant (blank) lines are badly indicated by `ut_coverage`
To fix that we somehow need to do one of those hings:
1) Use DBA_TRIGGERS for trigger source (that column is of type LONG) so it will cause issues.
2) Use DBA_TRIGGERS to identify the lines offset between line no in DBA_SOURCE and DBA_TRIGGERS
3) Figure out, how to identify the first line of coverage in TRIGGER BODY by algorithm, so that we dont need to scan DBA_TRIGGERS.

First two solutions will affect the performance of coverage gathering significantly.

**Comment by Shoelace** [comment]
i think the dba_triggers.trigger_body start with the first 'BEGIN' or 'DECLARE' or 'COMPOUND TRIGGER' statement.

but you are right. calculating teh offset migth be difficult.

here is my first attempt
```PL/SQL
SELECT owner,name, min(line)
FROM 
dba_source
WHERE 1=1
and type = 'TRIGGER'
and (text like '%BEGIN%' or text like '%DECLARE%' or text like '%COMPOUND%')
and regexp_like( text, '\w*BEGIN.*')
and name = '<TRIGGERNAME>'
group by owner,name
;
```

**Comment by jgebal** [comment]
Yup - works :)
Will need to add it to the `ut_coverage` code as:
```
             s.line - 
             coalesce(
               case when type!='TRIGGER' then 0 end,
               (select min(t.line) - 1
                  from dba_source t
                 where t.owner = s.owner and t.type = s.type and t.name = s.name
                   and regexp_like( t.text, '\w*(begin|declare|compound).*','i'))
             ) as line,
```
We joust need to filter on the `line >0`

**Comment by Pazus** [comment]
Remember about comments...

**Comment by jgebal** [comment]
Resolved by #229
